model/project.go

```diff
@@ -1932,29 +1945,31 @@ func (p *Project) BuildProjectTVPairsWithAlias(vars []ProjectAlias) ([]TVPair, [
 		}
 
 		for _, variant := range p.BuildVariants {
-			if isValidRegexOrTag(variant.Name, variant.Tags, v.VariantTags, variantRegex) {
-				for _, t := range p.Tasks {
-					if t.Patchable != nil && !(*t.Patchable) {
-						continue
-					}
-					if !isValidRegexOrTag(t.Name, t.Tags, v.TaskTags, taskRegex) {
-						continue
-					}
-
-					if p.FindTaskForVariant(t.Name, variant.Name) != nil {
-						pairs = append(pairs, TVPair{variant.Name, t.Name})
-					}
-				}
+			if !isValidRegexOrTag(variant.Name, variant.Tags, v.VariantTags, variantRegex) {
+				continue
+			}
 
-				if v.Task == "" {
+			for _, t := range p.Tasks {
+				if !isValidRegexOrTag(t.Name, t.Tags, v.TaskTags, taskRegex) {
 					continue
 				}
-				for _, displayTask := range variant.DisplayTasks {
-					if !taskRegex.MatchString(displayTask.Name) {
+
+				if bvtu := p.FindTaskForVariant(t.Name, variant.Name); bvtu != nil {
+					if bvtu.IsDisabled() || bvtu.SkipOnRequester(requester) {
 						continue
 					}
-					displayTaskPairs = append(displayTaskPairs, TVPair{variant.Name, displayTask.Name})
+					pairs = append(pairs, TVPair{variant.Name, t.Name})
+				}
+			}
+
+			if v.Task == "" {
```

Comment:
Why do we wait until right before display task construction to do this?

Reply:
This was how the logic was already, so I just left it. The original PR (#673) didn't make any comment about it, but if I had to guess, it's indirectly checking that [`taskRegex` is non-nil](https://github.com/evergreen-ci/evergreen/blob/c89980b58abab4449e6b5c374bc506cfaad29bbd/model/project_aliases.go#L538-L540) before using it to check display tasks. I changed it to check for a non-nil `taskRegex` to make it clearer.